src/utilities/device_utils.h

```diff
@@ -360,7 +360,6 @@ using hypre_DeviceItem = sycl::nd_item<3>;
    if (cudaSuccess != err) {                                                                 \
       printf("CUDA ERROR (code = %d, %s) at %s:%d\n", err, cudaGetErrorString(err),          \
                    __FILE__, __LINE__);                                                      \
-      hypre_assert(0); exit(1);                                                              \
```

Comment:
Why remove? In the case of CUDA ERROR, do we want to exit right away?

Reply:
@rfalgout and I talked about this at some point. My understanding was that we shouldn't exit from hypre. If confirmed, this behavior needs to be replicated in the other macros (in a separate PR)

Reply:
OK. Then, at least a `return hypre_error` should be called? IMHO, there's no reason to continue to when a CUDA error occurs.

Reply:
Agreed